### DISCUSSION TITLE
How to disable cpu or gpu profiling?

### DISCUSSION THREAD

**Discussion by char101** [plasma-umass/scalene#517] [Q&A] [answered]
Hi,

I tried to run scalene in Windows using this command

`scalene --cpu --memory --html -m mymodule`

But got this message:

```
NOTE: The GPU is currently running in a mode that can reduce Scalene's accuracy when reporting GPU utilization.
Run once as Administrator or root (i.e., prefixed with `sudo`) to enable per-process GPU accounting.
```

My application does not use the GPU so I want to disable it. From the command line help I found

```
  --cpu                 profile CPU time (default:  True )
  --cpu-only            profile CPU time (deprecated: use --cpu )
  --gpu                 profile GPU time and memory (default: True )
```

I find this confusing. If the default is `True` then what is the purpose of specifying `--cpu` or `--gpu`. Usually a boolean flag can be disabled by specifying `--no-gpu` or `--gpu=no`. I tried both form and it does not work.

**Comment by emeryberger** [comment, marked as answer]
Thanks for the note. The warning about the GPU will no longer be printed (this is now in the repo version). If you specify nothing, you get full profiling. If you specify any of the features, you get just the ones you specify (OK, not really, you always get CPU profiling right now). However, Windows does not currently support memory profiling.
